contracts/ThalesRoyale/ThalesRoyale.sol

```diff
@@ -232,9 +246,54 @@ contract ThalesRoyale is Owned, Pausable {
         rounds = _rounds;
     }
 
+
+    function setClaimTime(uint _claimTime) public onlyOwner {
+        claimTime = _claimTime;
+    }
+
+    function _populateWinnersAndThereRewards(address[] memory alivePlayers) internal {
+        require(finished, "Royale must be finished");
+        require(alivePlayers.length > 0, "There is no alive players left in Royale");
+
+        rewardPerPlayer = reward.div(alivePlayers.length);
+
+        for (uint i = 0; i < alivePlayers.length; i++) {
+            rewardCollected[alivePlayers[i]] = false;
+        }
```

Comment:
obsolete, they are false at the start. 
Previous comment was to delete this for loop. 
However based on Danijel comment, maybe you would need to make accumulation logic. But that can be done similarly as in the EscrowThales, without loops

Reply:
yeah also never do for loops in solidity (I know I have one but it was just for testing purposes and its a read method)